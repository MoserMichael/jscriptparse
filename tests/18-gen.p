
# test generator functions

def mygen(from,to) {
    while from < to {
        yield from
        from = from + 1
    }
}

for n mygen(1,10) {
    println("mygen - num: {n}")
}


def mygen2(from,to) {
    for n mygen(from, to) {
        yield n
    }
}

for n mygen2(1,10) {
    println("mygen2 - num: {n}")
}


# test generator from try-catch block

def mygen3(from,to) {
    try 
        for n mygen(from, to) {
            yield n
        }
    catch e 
        console.log("exception {e}")

}

for n mygen3(1,10) {
    println("mygen3 - num: {n}")
}




