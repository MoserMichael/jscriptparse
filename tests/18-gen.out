mygen - num: 1
mygen - num: 2
mygen - num: 3
mygen - num: 4
mygen - num: 5
mygen - num: 6
mygen - num: 7
mygen - num: 8
mygen - num: 9
mygen2 - num: 1
mygen2 - num: 2
mygen2 - num: 3
mygen2 - num: 4
mygen2 - num: 5
mygen2 - num: 6
mygen2 - num: 7
mygen2 - num: 8
mygen2 - num: 9
